src/pybind/mgr/cephadm/serve.py

```diff
@@ -111,6 +113,19 @@ def serve(self) -> None:
             self.log.debug("serve loop wake")
         self.log.debug("serve exit")
 
+    def _check_certificates(self) -> None:
+        cert = self.mgr.get_store('grafana_crt')
+        key = self.mgr.get_store('grafana_key')
```

Comment:
for my info, when do we actually expect this to come up? Afaik we generally generate the cert/key pair unless the user provides one and, when deploying grafana daemons, we overwrite them if they fail `verify_tls`. So this situation would require either the user provided an invalid cert/key pair (in which case, is having them redeploy the grafana daemon so we overwrite them with our generated ones really what we want to recommend?) or our strategy for generating the cert/key pair is broken (in which case redeploying it likely wouldn't help because we'll probably just generate a bad cert/key pair again). Just want to get your thoughts on what the expected case is for this, as I feel like I'm missing something.

Reply:
The idea behind the original feature req is to check periodically the validity of the provided Grafana certificates and raise a health alert if they aren't healthy (expired). 

I check the code and you are right: if cert/key are not correct, we just log a warning:

`logger.warning('Provided grafana TLS certificates invalid: %s', str(e))`

and re-generate our own certificates. To me this sounds a little bit weird specially if the user has provided its own certificates. In this case he won't see the warning and internally we overwrite his certificates. I think probably we should be more clever in this case and do not overwrite the certs if they weren't generated by us. In this case we should raise a health warning indicating the grafana certs are invalid. In case the certs were generated by us we can go and generate new ones.

WDYT?

Reply:
I think only overwriting if we generated them makes sense as long as we have a good way of verifying if the certs were generated by us or not. I see we set the organisation to "Ceph" and common name to "cephadm" when we generate our own self signed certs. Perhaps there's a way to check for this?

Reply:
I didn't go through the code but I +1 the idea to avoid overwriting the certificates if they're originally provided by the operator and even though self-signed fixes and maintain the wanted config/behavior, it should be problematic for an operator who is supposed to provide this info.
HEALTH_WARN with an appropriate message would help and doesn't hurt as long as the goal is to alert the operator and update the provided certificate.

Reply:
I added some logic and functionality to check the certificates issuer so we only overwrite them if they were generated by us (Ceph/cephadm). New changes seem to work but I'm not happy to duplicate the checks both in monitoring and in the main serve.py loop ... though I didn't find any simple way to avoid that as the monitoring code is only called when a user tries to reconfig/redeploy a daemon.